### PR TITLE
feat: Introduce flame_noise, deprecate NoiseEffectController

### PR DESCRIPTION
**Pull request by luanpotter**
# Description

* Deprecate the NoiseEffectController that is based on the deprecated vector2 impl
* Add a new bridge package `flame_noise` to bridge fast noise

Note: the *goal* of this PR is to allow the immediate deprecation of the old NoiseEffectController by providing an experimental, suitable replacement. I believe the package flame_noise will look nothing like this at all if I am able to land [some improvements to the underlying fast_noise lib](https://github.com/frankpepermans/fast_noise/pull/5). We will be able to have for example a generic NoiseEffectController.

However I believe we should merge this as an experimental package for now to unblock removing the current implementation from flame which is negatively affecting our scores on pub. Therefore I would advise we don't spend any time discussing specifics of how the API/impl provided for the flame_noise, and just go with _something_ instead of nothing.

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change (yet).

Not yet. Will remove deprecated noise class later.

## Related Issues

https://github.com/flame-engine/flame/issues/2343
https://github.com/flame-engine/flame/issues/2329

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

File: packages/flame/lib/src/effects/controllers/noise_effect_controller.dart [hunk 1]
```diff
@@ -21,7 +18,9 @@ import 'package:vector_math/vector_math_64.dart';
 /// example, putting into a `MoveEffect.by` will create a shake motion, where
 /// the magnitude and the direction of shaking is controlled by the effect's
 /// `offset`.
+@Deprecated('Use flame_noise instead')
```
Comment (spydon) [inline packages/flame/lib/src/effects/controllers/noise_effect_controller.dart:21]: Add which version that it will be removed in, which is v1.8.0

File: packages/flame/lib/src/effects/controllers/noise_effect_controller.dart [hunk 2]
```diff
@@ -21,7 +18,9 @@ import 'package:vector_math/vector_math_64.dart';
...
 class NoiseEffectController extends DurationEffectController {
+  @Deprecated('Use flame_noise instead')
```
Comment (spydon) [inline packages/flame/lib/src/effects/controllers/noise_effect_controller.dart:23]: I don't think you have to deprecate both the constructor and the class? I think the class should be enough.

File: packages/flame_noise/README.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# flame_noise
```
Comment (spydon) [inline packages/flame_noise/README.md:1]: Should this readme look the same as the other ones that you consolidated?

**Review by spydon — APPROVED** [review]

File: packages/flame/lib/src/effects/controllers/noise_effect_controller.dart (hunk 2 above)
Comment (luanpotter) [reply packages/flame/lib/src/effects/controllers/noise_effect_controller.dart:23]: I recall vscode complaining if the constructor wasn't. Let me try again

Comment (luanpotter) [reply packages/flame/lib/src/effects/controllers/noise_effect_controller.dart:23]: image

**Review by spydon — APPROVED** [review]
